compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2436,8 +2436,9 @@ impl<'tcx> Ty<'tcx> {
             },
 
             // "Bound" types appear in canonical queries when the
-            // closure type is not yet known
-            Bound(..) | Param(_) | Infer(_) => None,
+            // closure type is not yet known, and `Placeholder` and `Param`
```

Comment:
wait... how does `Bound` appear? shouldn't we map this back to infer vars again? Is it used in const eval which canonicalizes but never instantiates?

Reply:
Bound may no longer be needed 🤔 -- i think that was preexisting